pycytominer/tests/test_feature_select.py

```diff
@@ -73,6 +71,96 @@
+        noise_removal_perturb_groups=data_df_groups,
+        noise_removal_stdev_cutoff=2,
+    )
+    result3 = feature_select(
+        data_df,
+        features=data_df.columns.tolist(),
+        operation="noise_removal",
+        noise_removal_perturb_groups=data_df_groups,
+        noise_removal_stdev_cutoff=3.5,
+    )
+    expected_result1 = data_df[["x", "y"]]
+    expected_result2 = data_df[[]]
+    expected_result3 = data_df[["x", "y", "z", "zz"]]
+    pd.testing.assert_frame_equal(result1, expected_result1)
+    pd.testing.assert_frame_equal(result2, expected_result2)
+    pd.testing.assert_frame_equal(result3, expected_result3)
+
+    # Test on data_unique_test_df, which has 100 rows
+    data_unique_test_df_groups = []
+    # Create a 100 element list containing 10 replicates of 10 perturbations
+    for elem in ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]:
+        data_unique_test_df_groups.append([elem] * 10)
+    # Unstack so it's just a single list
+    data_unique_test_df_groups = [
+        item for sublist in data_unique_test_df_groups for item in sublist
+    ]
+
+    result4 = feature_select(
+        data_unique_test_df,
+        features=data_unique_test_df.columns.tolist(),
+        operation="noise_removal",
+        noise_removal_perturb_groups=data_unique_test_df_groups,
+        noise_removal_stdev_cutoff=3.5,
+    )
+    result5 = feature_select(
+        data_unique_test_df,
+        features=data_unique_test_df.columns.tolist(),
+        operation="noise_removal",
+        noise_removal_perturb_groups=data_unique_test_df_groups,
+        noise_removal_stdev_cutoff=500,
+    )
+    expected_result4 = data_unique_test_df[["a", "b"]]
+    expected_result5 = data_unique_test_df[["a", "b", "c", "d"]]
+    pd.testing.assert_frame_equal(result4, expected_result4)
+    pd.testing.assert_frame_equal(result5, expected_result5)
+
+    # Test the same as above, except that data_unique_test_df_groups is now made into a metadata cpl
+    data_unique_test_df2 = data_unique_test_df.copy()
+    data_unique_test_df2["perturb_group"] = data_unique_test_df_groups
+    result4b = feature_select(
+        data_unique_test_df2,
+        features=data_unique_test_df.columns.tolist(),
+        operation="noise_removal",
+        noise_removal_perturb_groups="perturb_group",
+        noise_removal_stdev_cutoff=3.5,
+    )
+    result5b = feature_select(
+        data_unique_test_df2,
+        features=data_unique_test_df.columns.tolist(),
+        operation="noise_removal",
+        noise_removal_perturb_groups="perturb_group",
+        noise_removal_stdev_cutoff=500,
+    )
+    expected_result4b = data_unique_test_df2[["a", "b", "perturb_group"]]
+    expected_result5b = data_unique_test_df2[["a", "b", "c", "d", "perturb_group"]]
+    pd.testing.assert_frame_equal(result4b, expected_result4b)
+    pd.testing.assert_frame_equal(result5b, expected_result5b)
+
+    # Test assertions based on type of perturbation groups passed
```

Comment:
4a and 5b are the same tests as 4 and 5, but I pass in a string for the metadata column instead of passing in metadata directly.

I should have removed that comment. My original intention was to write tests to check for assertion failure like when you pass in invalid perturbation, but I noticed the other tests didn't test for those nor was "import unittest" used (seems like using the unittest framework is the only way to test assertion errors). I think it's a good ideal to test for these anyways especially in light of your comment above. Can I import unittests for this?